projects/core/src/occ/adapters/cms/occ-cms-page.adapter.ts

```diff
@@ -44,16 +49,34 @@ export class OccCmsPageAdapter implements CmsPageAdapter {
    */
   load(pageContext: PageContext): Observable<CmsStructureModel> {
     const params = this.getPagesRequestParams(pageContext);
+    if (this.featureConfigService.isEnabled('newCmsEndpoint')) {
+      return this.userIdService.getUserId().pipe(
+        switchMap((userId: string) => {
+          const endpoint = !pageContext.type
+            ? this.occEndpoints.buildUrl('page', {
+                urlParams: { id: pageContext.id, userId },
+              })
+            : this.occEndpoints.buildUrl('pages', {
+                urlParams: { userId },
+                queryParams: params,
+              });
 
-    const endpoint = !pageContext.type
-      ? this.occEndpoints.buildUrl('page', {
-          urlParams: { id: pageContext.id },
-        })
-      : this.occEndpoints.buildUrl('pages', { queryParams: params });
-
-    return this.http
-      .get(endpoint, { headers: this.headers })
-      .pipe(this.converter.pipeable(CMS_PAGE_NORMALIZER));
+          return this.http.get(endpoint, { headers: this.headers });
+        }),
+        this.converter.pipeable(CMS_PAGE_NORMALIZER)
+      );
+    } else {
+      const endpoint = !pageContext.type
```

Comment:
Hopefully, we can refactor this section of code to remove it from the else statement. If the condition `this.featureConfigService.isEnabled('newCmsEndpoint')` evaluates to true, the function will return a value without proceeding further. Otherwise, if the condition is false, the if block will be skipped, and a different value will be returned.

There is no need to nesting a code if it's not necessary.

Reply:
sure, fixed